src/main/java/uk/gov/hmcts/reform/wataskmanagementapi/taskconfiguration/services/calendar/PublicHolidaysCollection.java

```diff
@@ -0,0 +1,46 @@
+package uk.gov.hmcts.reform.wataskmanagementapi.taskconfiguration.services.calendar;
+
+import feign.Feign;
+import feign.codec.Decoder;
+import feign.codec.Encoder;
+import org.springframework.cache.annotation.Cacheable;
+import org.springframework.stereotype.Component;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+/**
+ * Stores all public holidays for england and wales retrieved from Gov uk API: https://www.gov.uk/bank-holidays/england-and-wales.json .
+ */
+@Component
+public class PublicHolidaysCollection {
+
+    public Decoder feignDecoder;
+    public Encoder feignEncoder;
+
+    public PublicHolidaysCollection(Decoder feignDecoder, Encoder feignEncoder) {
+        this.feignDecoder = feignDecoder;
+        this.feignEncoder = feignEncoder;
+    }
+
+    @Cacheable(value = "public_holidays_uri_cache", key = "#uri", sync = true)
```

Comment:
Current cache time out is set to 30 minutes. Do you want to use the same for this cache as well? I recon we can have much higher value for this cache as calendar would not changed regularly

Reply:
there is separate ticket for cache, will take care in that